### PR TITLE
[ISSUE-109] Attachement should be VertxGen object

### PR DESCRIPTION
**Pull request by gaol**
Issue: https://github.com/vert-x3/vertx-mail-client/issues/109

Changes in PR:
- Changed `MailAttachment` from `class` to `interface`
   - It provides 3 `create` methods to correspond the 3 constructors. 
> NOTE: This is a break change compare to `3.x`.
- Remove generated `*Converter` classes
   - It actually got removed when rebuilding the project after update.
- Update unit tests to use new `MailAttachment` API.

### CODE REVIEW

**Comment by vietj** [comment]
this looks great, I will merge it soon.

**Comment by vietj** [comment]
I will document this as breaking change in the wiki

File: src/main/java/io/vertx/ext/mail/Utils.java [hunk 1]
```diff
@@ -28,26 +28,26 @@
  * @author <a href="http://oss.lehmann.cx/">Alexander Lehmann</a>
  *
  */
-class Utils {
+public class Utils {
```
Comment (vietj) [inline src/main/java/io/vertx/ext/mail/Utils.java]: Move this class to the `impl` package if you need to make it public

**Comment by gaol** [comment]
@vietj Done by moving the `Utils` methods into `impl` package. :)

**Comment by vietj** [comment]
Updated the wiki https://github.com/vert-x3/wiki/wiki/4.0.0-Deprecations-and-breaking-changes/_compare/cdb6ba4e732cec9022bf3437468e88dcfe1b9c06...1b9615704f1bd168f1ad5ddec21d67a6ebfe4076

**Comment by vietj** [comment]
thanks @gaol for the contribution

**Pull request merged**